app/scripts/metamask-controller.js

```diff
@@ -4315,6 +4321,48 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Stops exposing the specified chain ID to all third parties.
+   * Exposed chain IDs are stored in caveats of the permittedChains permission. This
+   * method uses `PermissionController.updatePermissionsByCaveat` to
+   * remove the specified chain ID from every permittedChains permission. If a
+   * permission only included this chain ID, the permission is revoked entirely.
+   *
+   * @param {string} targetChainId - The chain ID to stop exposing
+   * to third parties.
+   */
+  removeAllChainIdPermissions(targetChainId) {
+    this.permissionController.updatePermissionsByCaveat(
+      CaveatTypes.restrictNetworkSwitching,
+      (existingChainIds) =>
+        CaveatMutatorFactories[
+          CaveatTypes.restrictNetworkSwitching
+        ].removeChainId(targetChainId, existingChainIds),
+    );
+  }
+
+  removeNetworkConfiguration(networkConfigurationId) {
```

Comment:
This whole function will probably change following https://github.com/MetaMask/core/pull/4286. There will only ever be one network configuration associated with a chain. `removeNetworkConfiguration` will be replaced with `removeNetwork` and take a `chainId` rather than a `networkConfigurationId`. So we may simply be able to say:

``` typescript
removeNetwork(chainId) {
  this.removeAllChainIdPermissions(chainId);
  this.networkController.removeNetwork(chainId);
}
```